drivers/include/periph/wdt.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *           (C) 2019 Inria
+ *           (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_wdt WDT
+ * @ingroup     drivers_periph
+ * @brief       watchdog timer peripheral driver
+ *
+ * @{
+ *
+ * @file        wdt.h
+ * @brief       watchdog peripheral interface definitions
+ *
+ * @author      Thomas Geithner <thomas.geithner@dai-labor.de>
+ *              Francisco Molina <francois-xavier.molina@inria.fr>
+ *              Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef PERIPH_WDT_H
+#define PERIPH_WDT_H
+
+#include <stdint.h>
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Signature for the watchdog early warning callback
+ *
+ * @param[in] arg           optional argument which is passed to the
+ *                          callback
+ */
+typedef void(*wdt_cb_t)(void *arg);
+
+/**
+ * @brief    Start watchdog timer
+ */
+void wdt_start(void);
+
+/**
+ * @brief    Stop watchdog timer
+ */
+void wdt_stop(void);
+
+/**
+ * @brief    Reset the watchdog timer counter, delay system reset
+ */
+void wdt_kick(void);
+
+/**
+ * @brief    Set up the wdt timer, only use max_time if normal operation
+ *           set min_time and max_time for windowed timer.
+ *
+ * @param[in] min_time       lower bound for windowed watchdog in us
```

Comment:
Looking at datasheet for other platforms max timeout can vary. It goes up to some seconds for ATMEGA, STM32. Hours for MSP430, nd for KINETIS since it has a 32bit counter running from the LPO (1Khz) the max timeout can be set to many days. NRF can go up to 50hrs approx. For ESP I'm not very clear... seems to depend if its clocked from the RTC or BUS clock and goes between ms and s.

In that case using uin32_t for us would be a constraint since the max achievable time would be a little more than 1 hour and since platforms that can go under ms only go around 100us, it woyld make sense to use ms as the base unit instead of us.

Reply:
I really doubt anyone would need µs precision for the watchdog.
Using ms here really seems like the most sensible approach.